### PR TITLE
remove unpin from body types

### PR DESCRIPTION
**Pull request by robjtede**
## PR Type

Improvement

## PR Checklist

- [x] Tests for the changes have been added / updated.
- [x] Documentation comments have been added / updated.
- [x] A changelog entry has been made for the appropriate packages.
- [x] Format code with the latest stable rustfmt.
- [x] (Team) Label with affected crates and semver status.

## Overview
- `impl<T: MessageBody> MessageBody for Pin<Box<T>>`
- The type parameter of `Response` no longer has a default.
- The `Message` variant of `body::Body` is now `Pin<Box<dyn MessageBody>>`.
- `BodyStream` and `SizedStream` are no longer restricted to Unpin types.

### CODE REVIEW

File: actix-http/src/lib.rs [hunk 1]
```diff
@@ -6,7 +6,6 @@
 //! | `openssl`        | TLS support via [OpenSSL].                            |
 //! | `rustls`         | TLS support via [rustls].                             |
 //! | `compress`       | Payload compression support. (Deflate, Gzip & Brotli) |
-//! | `secure-cookies` | Adds for secure cookies. Enables `cookies` feature.   |
```
Comment (therealprof) [inline actix-http/src/lib.rs:9]: Doesn't look like it belongs here?

Comment (robjtede) [reply actix-http/src/lib.rs:9]: was missed from previous PR

**Comment by therealprof** [comment]
Sweet!

Any indication of the performance impact? It sounds like the ability to use `mem::replace` might be a performance improvement but magic behind the scenes could negate that. 😅

File: actix-http/src/lib.rs (hunk 1 above)
Comment (therealprof) [reply actix-http/src/lib.rs:9]: I suspected so but I'm not a big fan of throwing random stuff into an otherwise clean PR. 😅

**Review by therealprof — APPROVED** [review]

**Comment by robjtede** [comment]
> Any indication of the performance impact? It sounds like the ability to use mem::replace might be a performance improvement...

The main impact of this change is to accept more stream types and reduce boxes needed which is a small improvement.

**Comment by therealprof** [comment]
> The main impact of this change is to accept more stream types and reduce boxes needed which is a small improvement.

Whether removing boxes is an improvement always depends on the data contained within and how it is used. A `Box` can also provide a performance improvement for large datastructures otherwise passed around by value.

I guess we will see the overall performance and memory use impact when we run it through the paces. 😅

CC @thalesfragoso

**Pull request merged**